[Issue #169] Allow plugins to be fetched behind proxy

DeanBrunt opened the pull request:
## Description
<!-- Please include a summary of the changes, Please add any additional motivation and context as needed -->
This change adds support for passing a proxy server to Grafana 
and the plugins init container, such that plugins can be fetched
from behind a proxy.

## Relevant issues/tickets
<!-- Please include a link to the issue or ticket that applies to this pull request, delete this heading if not relevant -->

Resolves #169 

## Type of change

<!-- Please delete options that are not relevant. -->
- [X] New feature (non-breaking change which adds functionality)

## Checklist
<!-- Tick options that apply, in-code tests are not required but please provide test cases and list steps in "verification steps" with the steps you used to verify this -->
- [X] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

## Verification steps
<!-- Please include a series of steps to verify that the functionality/bug fix works, ideally the steps you used to test these changes(if applicable, if not, delete this section)-->
- Deploy a forward proxy that will be accessible from the Kubernetes cluster that the Grafana will be deployed into (I used `cproxy`)
- Deploy Grafana Operator into said cluster
- Deploy a Grafana instance with configuration under the `deployment` section that is as follows:
```
httpProxy:
    enabled: true
    url: <proxy_url>
```
- Deploy a dashboard with plugins to the instance
- You should observe plugin fetches succeed and (if the proxy emits logs), logs on the proxy indicating it has been responsible for resolving the requests.
- Optionally: Disable all egress traffic to the internet except if it goes via the proxy (simulates the environment I would typically expect this to be deployed in)

HubertStefanski commented:
@DeanBrunt really happy to see this being worked on, we had plenty of requests/questions about this :+1:

DeanBrunt commented:
Ok, I've managed to test this now and I believe it does what is expected.
I'll rebase this PR and then take it out of draft.

DeanBrunt commented:
Not entirely sure to tempt the Prow bot to remove the WIP and rebase labels...
Expected that to happen when I ditched WIP from the title and rebased and squashed.
@HubertStefanski any ideas?

pb82 commented:
@DeanBrunt issues with lint have been fixed in another PR and yours does not seem to cause any conflicts. I'm going to merge this, change looks good!
